### ISSUE TITLE
rule proposal: special-param-shadow integer and float declarations

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#157]
Parent: #72

## Rule ID and name

Extension of `compat/special-param-shadow`, Shadowing special shell parameters.

## Proposed behavior

Recognize Zsh `integer` and `float` local declaration commands when they declare a curated shell-set parameter, while preserving the existing declaration option, global-mode, query-mode, and suppression contracts.

## Bad and good examples

```zsh
# Bad
integer status=0

# Good
integer command_status=0
```

## Severity

Warning, matching the existing rule. A local declaration can replace shell-managed behavior in the current dynamic scope and nested calls.

## Evidence review

The current six-repository, 20-file reference corpus contains no `integer` or `float` declaration of the curated special parameters. A full local z-shell organization search and a broader GitHub code search for representative exact forms such as `integer ZSH_VERSION`, `integer OSTYPE`, `integer status`, and their `float` equivalents also found no qualifying example.

This absence does not change Zsh semantics. It means the proposal does not meet the repository rule policy's real-code evidence gate.

## False-positive boundary

- Match only a statically known `integer` or `float` command in a Zsh source profile.
- Parse static option words and exclude explicit global, query, display, and function modes consistently with the existing rule.
- Do not treat an ordinary command argument containing assignment-like text as a declaration.
- Preserve deliberate compatibility-shim suppression with a reason.

## Parser dependency

`mvdan/sh` currently represents these forms as `*syntax.CallExpr`, not `*syntax.DeclClause`. An implementation would need a focused static command/argument path plus native-valid and closely related native-invalid fixtures. Native Zsh remains semantic authority.

## Suppression

Use the existing directive:

```zsh
# zsh-lint disable=compat/special-param-shadow -- intentional compatibility shim
```

## Priority and effort

Priority P3, effort S if qualifying evidence appears.

## Disposition

Not planned because the required real-code evidence is absent. Reopen only with a qualifying production example and rerun the false-positive analysis against the then-current configured corpus.
